client/src/components/StreamField/blocks/BaseSequenceBlock.js

```diff
@@ -2,23 +2,111 @@
 
 /* global $ */
 
+import EventEmitter from 'events';
 import { escapeHtml as h } from '../../../utils/text';
 
-export class BaseSequenceChild {
-  constructor(blockDef, placeholder, prefix, index, id, initialState, opts) {
+class ActionButton {
+  constructor(sequenceChild) {
+    this.sequenceChild = sequenceChild;
+  }
+
+  render(container) {
+    const label = this.sequenceChild.strings[this.labelIdentifier] || this.labelIdentifier;
+
+    this.dom = $(`
+      <button type="button" class="c-sf-block__actions__single" title="${h(label)}">
+        <i class="icon icon-${h(this.icon)}" aria-hidden="true"></i>
```

Comment:
question: in light of #6107, would this not be the perfect place to make the switch for the StreamField action buttons? Doing it now would mean less end-dev maintenance

Reply:
Makes sense, yep... I hadn't made the connection that this was still using the icon font! I guess that involves coming up with some new infrastructure for passing SVGs to front-end code though - maybe something like the 'strings' dict for translations. (Which, come to think of it, is also not brilliantly placed to serve 'pluggable' mechanisms like this, where the third-party code will want to use strings / icons outside of the set that's been [picked out by the core code](https://github.com/wagtail/wagtail/blob/9f3e6585afd07bab15c22257ecf4ed4eab5ab0dd/wagtail/core/blocks/stream_block.py#L593-L598).)

Reply:
it does indeed. the regular templates have `{% icon name=".." %}` which uses https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templates/wagtailadmin/shared/icon.html

I suppose adding an icon class similar to https://github.com/wagtail/wagtail/blob/main/client/src/components/Icon/Icon.tsx would solve that bit.